operator-pipeline-images/operatorcert/static_tests/community/bundle.py

```diff
@@ -27,17 +28,66 @@
     validate_timestamp,
 )
 
+# convert table for OCP <-> k8s versions
+# for now these are hardcoded pairs -> if new version of OCP:k8s is released,
+# this table should be updated
+K8S_TO_OCP = {
+    "4.6": "1.19",
+    "4.7": "1.20",
+    "4.8": "1.21",
+    "4.9": "1.22",
+    "4.10": "1.23",
+    "4.11": "1.24",
+    "4.12": "1.25",
+    "4.13": "1.26",
+    "4.14": "1.27",
+    "4.15": "1.28",
+    "4.16": "1.29",
+}
+
 
 class GraphLoopException(Exception):
     """
     Exception raised when a loop is detected in the update graph
     """
 
 
+def process_ocp_version(ocp_metadata_version: Any) -> Any:
+    """
+    Helper function to process the OCP version.
+    OCP version can be either range of mim-max supported versions
+    (v4.7-v4.9), range of up to max supported version (v4.10)
+    or exact single version (=v4.9)
+    """
+    # we are calling indices to extract the supported OCP version
+    # which is then used in operator-sdk bundle validate
+    # to run API deprecation test
+    indices_url = "https://catalog.redhat.com/api/containers/v1/operators/indices"
+    params = {
+        "filter": "organization==community-operators",
+        "sort_by": "ocp_version[desc]",
+        "include": "data.ocp_version",
+        "ocp_versions_range": ocp_metadata_version,
+    }
+    # timeout param set in order to not hang indefinitely
+    rsp = requests.get(indices_url, params=params, timeout=30)
+    get_indices = rsp.json().get("data")
+
+    ocp_version = get_indices[0].get("ocp_version") if get_indices else "4.16"
```

Comment:
The default here is I think not correct. If the API doesn't return anything the test should fail in my opinion.

Reply:
I was thinking about it, based on the info in jira, if target version is not specified, test should be executed against all supported versions. Since that would be IMO unnecessary I wanted to execute test against the latest OCP version we have. What we can do if the user does not specify annotation is either fail the test entirely or fallback to the latest supported version and test it against it. I think you are right and strict approach might be better.

Reply:
Based on the discussion via chat and also as stated in [ISV-4290](https://issues.redhat.com/browse/ISV-4290) in case the annotation is missing, we would test all supported versions - which means 4.16:1.29 (current OCP:k8s) version would be used as a default value for the test. The test is executed retrospectively, so it would test also all past deprecations/supported ocp versions.